assets/common/src/model_deployment/deploy.py

```diff
@@ -162,20 +162,33 @@ def parse_args():
 
 
 def get_ml_client():
-    """Return ML client."""
-    credential = AzureMLOnBehalfOfCredential()
+    """Return ML Client."""
+    has_obo_succeeded = False
     try:
+        credential = AzureMLOnBehalfOfCredential()
         # Check if given credential can get token successfully.
         credential.get_token("https://management.azure.com/.default")
+        has_obo_succeeded = True
     except Exception as ex:
-        # Fall back to ManagedIdentityCredential in case AzureMLOnBehalfOfCredential not work
-        print(f"Failed to get obo credentials - {ex}")
-        msi_client_id = os.environ.get("DEFAULT_IDENTITY_CLIENT_ID")
-        credential = ManagedIdentityCredential(client_id=msi_client_id)
-    try:
-        credential.get_token("https://management.azure.com/.default")
-    except Exception as ex:
-        raise (f"Failed to get credentials : {ex}")
+        # Fall back to ManagedIdentityCredential in case AzureMLOnBehalfOfCredential does not work
+        print(f"Failed to get OBO credentials - {ex}")
```

Comment:
Could we also add details on how to run component using OBO?

Reply:
Also if it make sense, could we create a notebook if not already existing to create MSI and attach to compute with sufficient enough permissions to perform registration and deployment in ws? Same for OBO

and then link them in the Exception we throw. 

cc: @skasturi